src/services/client-data-mapping-service.js

```diff
@@ -154,6 +156,48 @@ class ClientDataMappingService {
     });
   }
 
+  _mapNotificationEventParams(eventType, eventParams, allowedDocumentsById) {
+    switch (eventType) {
+      case EVENT_TYPE.revisionCreated:
+      case EVENT_TYPE.commentCreated:
+        return {
+          document: allowedDocumentsById.get(eventParams.documentId) || null
+        };
+      default:
+        throw new Error(`Unsupported event type '${eventType}'`);
+    }
+  }
+
+  async mapUserNotificationGroups(notificationGroups, user) {
+    const occurringDocumentIds = [...new Set(notificationGroups.map(g => g.eventParams.documentId).filter(x => x))];
+
+    const [occurringDocuments, allowedRooms] = await Promise.all([
+      this.documentStore.getDocumentsMetadataByIds(occurringDocumentIds),
+      this.roomStore.getRoomsOwnedOrJoinedByUser(user)
+    ]);
+
+    const allowedRoomsById = allowedRooms.reduce((map, item) => {
+      map.set(item._id, item);
+      return map;
+    }, new Map());
+
+    const allowedDocumentsById = occurringDocuments.reduce((map, item) => {
+      // The room might be not accessible to the user anymore:
+      if (!item.roomId || allowedRoomsById.has(item.roomId)) {
```

Comment:
Instead of a comment it's probably nicer to let the code speak here through a variable `isRoomStillAccessibleToUser = allowedRoomsById.has(item?.roomId)`

Reply:
The code speaks now :-)

Reply:
In very eloquent language 👌